### ISSUE TITLE
Ubuntu 22.04 on arm64 won't install newer than version 0.26.0 from apt repository

### ISSUE THREAD

**Issue by timnolte** [vectordotdev/vector#20339]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

I updated my server to use the new `apt.vector.dev` repository and my server is still stuck with only being able to install version 0.26.0. When I ran the installer for the latest version of Vector it converted my `*.toml` file to a `*.yaml` but because the binary installed ended up being 0.26.0 it meant the services couldn't start up since the `*.toml` file was "missing". It seems there is some problem with apt reading the new repository correctly.

### Version

vector 0.26.0 (aarch64-unknown-linux-gnu c6b5bc2 2022-12-05)

### Additional Context

`/etc/apt/sources.list.d/vector.list` contents:

```
deb [signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg] https://apt.vector.dev/ stable vector-0
```

```shell
$ sudo lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy
```

```shell
$ sudo hostnamectl
...
Operating System: Ubuntu 22.04.4 LTS
          Kernel: Linux 6.5.0-1021-oracle
    Architecture: arm64
```

**Comment by jszwedko** [comment]
Hi @timnolte ,

I'm having trouble reproducing this on a fresh Ubuntu 22.04 arm64 system (it installs v0.37.1 as expected). Can you share the output of `apt-cache show vector`? Also, is it installing 0.26.0 from the new repositories? That might indicate you have some pinning somewhere.

**Comment by timnolte** [comment]
@jszwedko so I checked the `apt-cache show vector` results and I didn't see anything unusual. However, I checked `/etc/apt/preferences.d/` and sure enough there was a `vector` file there that had the version pinned. I have no idea why I would have pinned anything to that version, obviously I have had this installed for quite awhile. I removed the version pinning file and was able to get the latest version of vector installed.
